mh/pat.py

```diff
@@ -1755,15 +1823,40 @@ def sendNtcFriendAdd(self, recipient_id, accepted, seq):
         """
         if not accepted:
             return
-        data = b""
-        data += pati.lp2_string(recipient_id)
+
+        users = self.session.find_users(recipient_id, "", 1, 1)
+        if len(users) == 0:
+            self.sendAnsAlert(PatID4.NtcFriendAdd, "<LF=8><BODY><CENTER>Player is Offline.<END>", seq)
+            return
+        partner_session = users[0]
+
+        # Tell sender their friend request was accepted
+        data_1 = b""
+        data_1 += pati.lp2_string(recipient_id)
+        friend = pati.FriendData()
+        friend.index = pati.Long(2)
+        friend.capcom_id = pati.String(self.session.capcom_id)
+        friend.hunter_name = pati.String(self.session.hunter_name)
+        data_1 += friend.pack()
+        data_1 += struct.pack(">B", accepted)
+        partner_pat_handler = self.server.get_pat_handler(partner_session)
+        if partner_pat_handler is None:
+            return
+        partner_pat_handler.send_packet(PatID4.NtcFriendAdd, data_1, seq)
```

Comment:
This time it does need to stay that way, because the `return` skips the whole "tell accepter that their acceptance was acknowledged" block that follows (which is the intended behavior).